app/src/db/models/document.py

```diff
@@ -46,3 +46,9 @@ class Chunk(Base, IdMixin, TimestampMixin):
 class ChunkWithScore:
     chunk: Chunk
     score: float
+
+
+@dataclass
+class DocumentWithMaxScore:
+    document: Document
+    max_score: float
```

Comment:
Reminder to self: Since retrieved chunks can come from the same document, a document can have multiple scores. `max_score` is the maximum of those scores across the chunks.

Reply:
Added a comment!